### PR TITLE
Error in Put transaction in chaos environment

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**
### Changes
- does not fail send txn immediately with a non success status code

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Review by peterlimg — APPROVED** [review]
lgtm

File: zcncore/transaction_base.go [hunk 1]
```diff
@@ -361,6 +359,14 @@ func (t *Transaction) submitTxn() {
 				return
 			}
 
+			if res.StatusCode != http.StatusOK {
```
Comment (peterlimg) [inline zcncore/transaction_base.go:364]: ah, giving none `StatusOK` means the txn execution failed due to some reason("invalid txn nonce" for example), the txn is actually reached to the miners, and we should return the error to the caller through the `resultC` channel. 

Shouldn't the error returned by`res, err := req.Post()` give the clue about the chaos caused node unreachable info? and I think it should have been handled above.

**Review by peterlimg — CHANGES_REQUESTED** [review]
lgtm

**Comment by peterlimg** [comment]
I'll close this PR as it will hide the true txn failure errors, the chaos caused txn submit failure case should have been handled, but it may slow as we will wait for timeout if all miners were offline. Feel free to re-open if needed.

File: zcncore/transaction_base.go (hunk 1 above)
Comment (Kishan-Dhakan) [reply zcncore/transaction_base.go:364]: No, on unreachable nodes status is 400 and req.Post() doesn't return any error.

**Pull request closed without merging**